Lib/xml/etree/ElementPath.py

```diff
@@ -253,13 +254,17 @@ def select(context, result):
                 if elem.get(key) is not None:
                     yield elem
         return select
-    if signature == "@-='":
-        # [@attribute='value']
+    if signature == "@-='" or signature == "@-!='":
+        # [@attribute='value'] or [@attribute!='value']
         key = predicate[1]
         value = predicate[-1]
+        negate_predicate = '!' in signature
         def select(context, result):
             for elem in result:
-                if elem.get(key) == value:
+                matches = elem.get(key) == value
+                if negate_predicate:
+                    matches = not matches
```

Comment:
This doesn't look correct. If the element does not have such an attribute, `.get()` will return `None`, which probably does not compare equal to `value`. Negating that gives `True`. However, `@attr != "value"` should only match if there is an attribute `attr` that compares unequal.

Reply:
Aah good catch, thank you! Fixed.